aquadoggo/src/server.rs

```diff
@@ -64,33 +35,40 @@ pub fn build_server(state: ApiState) -> Router {
         )
         // Add middlewares
         .layer(cors)
-        // Add shared state
-        .layer(Extension(state))
+        // Add shared context
+        .layer(Extension(context))
 }
 
 /// Start HTTP server.
-pub async fn start_server(config: &Configuration, state: ApiState) -> anyhow::Result<()> {
-    let http_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), config.http_port);
-    let server = build_server(state);
-    axum::Server::bind(&http_address)
-        .serve(server.into_make_service())
+pub async fn http_service(context: Context, signal: Shutdown, _tx: ServiceSender) -> Result<()> {
+    let http_port = context.config.http_port;
+    let http_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), http_port);
+
+    axum::Server::try_bind(&http_address)?
+        .serve(build_server(context).into_make_service())
+        .with_graceful_shutdown(async {
+            signal.await.ok();
+        })
         .await?;
+
     Ok(())
 }
```

Comment:
Just checking, but this is the actual service function we add to the service manager, all services will be passed as functions with the same signature as this. Correct?

Reply:
Yes exactly!